src/lambda_handler.py

```diff
@@ -1,24 +1,157 @@
-"""Simple AWS Lambda handler to verify functionality."""
+"""AWS Lambda handler to publish egress IPs from a provided list of AWS accounts."""
 
 # Standard Python Libraries
 from datetime import datetime, timezone
+from ipaddress import collapse_addresses, ip_network
 import logging
 import os
-from typing import Any, Optional, Union
+import re
+from typing import Any, Dict, Iterator, List, Optional, Set, Tuple, TypedDict, Union
 
 # Third-Party Libraries
-import cowsay
-import cowsay.characters
-
-# cisagov Libraries
-from example import example_div
+import boto3
 
 default_log_level = "INFO"
 logger = logging.getLogger()
 logger.setLevel(default_log_level)
 
 
-def failed_task(result: dict[str, Any], error_msg: str) -> None:
+def assume_role(role_arn: str, session_name: str) -> Tuple[str, str, str]:
+    """Assume the given role and return a tuple containing the assumed role's credentials."""
+    # Create an STS session with current credentials
+    sts: boto3.client = boto3.client("sts")
+
+    # Assume the provided role
+    response: Dict[str, Any] = sts.assume_role(
+        RoleArn=role_arn, RoleSessionName=session_name
+    )
+
+    return (
+        response["Credentials"]["AccessKeyId"],
+        response["Credentials"]["SecretAccessKey"],
+        response["Credentials"]["SessionToken"],
+    )
+
+
+def create_assumed_aws_client(
+    aws_service: str, role_arn: str, session_name: str
+) -> boto3.client:
+    """Assume the given role and return an AWS client for the given service using that role."""
+    role_credentials: Tuple[str, str, str] = assume_role(role_arn, session_name)
+
+    return boto3.client(
+        aws_service,
+        aws_access_key_id=role_credentials[0],
+        aws_secret_access_key=role_credentials[1],
+        aws_session_token=role_credentials[2],
+    )
+
+
+def create_assumed_aws_resource(
+    aws_service: str, region: str, role_arn: str, session_name: str
+) -> boto3.resource:
+    """Assume the given role and return an AWS resource object for the given service using that role."""
+    role_credentials: Tuple[str, str, str] = assume_role(role_arn, session_name)
+
+    return boto3.resource(
+        aws_service,
+        region_name=region,
+        aws_access_key_id=role_credentials[0],
+        aws_secret_access_key=role_credentials[1],
+        aws_session_token=role_credentials[2],
+    )
+
+
+def convert_tags(aws_resource: boto3.resource) -> Dict[str, str]:
+    """Convert resource tags from an AWS dictionary into a Python dictionary."""
+    try:
+        tags: Dict[str, str] = {x["Key"]: x["Value"] for x in aws_resource.tags}
+    except TypeError:
+        # This happens if there are no tags associated with the resource
+        tags = {}
+    return tags
+
+
+def get_ec2_ips(
+    ec2: boto3.resource, application_tag_name: str, publish_egress_tag_name: str
+) -> Iterator[Tuple[str, str]]:
```

Comment:
Again, I think a dictionary or named tuple might be better here.

Reply:
Please see https://github.com/cisagov/publish-egress-ip-lambda/pull/4/commits/a9ec40caa6dbf52a8c2f10e9afd6942702c52de4.